ci/matrix.yaml

```diff
@@ -1,384 +1,284 @@
+  # CUB uses `lid` to indicate launch strategies: whether CUB algorithms are:
+  # - launched from the host (lid0):
+  test_lid0:  { name: 'HostLaunch',   gpu: true, needs: 'build', invoke: { prefix: 'test', args: '-lid0'} }
+  # - launched from the device (lid1):
+  test_lid1:  { name: 'DeviceLaunch', gpu: true, needs: 'build', invoke: { prefix: 'test', args: '-lid1'} }
+  # - captured in a CUDA graph for deferred launch (lid2):
+  test_lid2:  { name: 'GraphCapture', gpu: true, needs: 'build', invoke: { prefix: 'test', args: '-lid2'} }
+
+  # Thrust:
+  test_cpu: { name: 'TestCPU', gpu: false, needs: 'build', invoke: { prefix: 'test', args: '-cpu-only'} }
+  test_gpu: { name: 'TestGPU', gpu: true,  needs: 'build', invoke: { prefix: 'test', args: '=gpu-only'} }
+
+# Project have the following properties:
+#
+# Keys are project subdirectories names. These will also be used in script names.
+#
+# - stds: A list of C++ standards to test. Required.
+# - name: The human-readable name of the project. Default is the project key.
+# - job_map: Map general jobs to arrays of project-specific jobs.
+#            Useful for things like splitting cpu/gpu testing for a project.
+#            E.g. "job_map: { test: ['test_cpu', 'test_gpu'] }" replaces
+#            the "test" job with distinct "test_cpu" and "test_gpu" jobs.
+projects:
+  cccl:
+    name: 'CCCL'
+    stds: [11, 14, 17, 20]
+  libcudacxx:
+    name: 'libcu++'
+    stds: [11, 14, 17, 20]
+  cub:
+    name: 'CUB'
+    stds: [11, 14, 17, 20]
+    job_map: { test: ['test_nolid', 'test_lid0', 'test_lid1', 'test_lid2'] }
+  thrust:
+    name: 'Thrust'
+    stds: [11, 14, 17, 20]
+    job_map: { test: ['test_cpu', 'test_gpu'] }
+  cudax:
+    stds: [17, 20]
+
+# testing -> Runner with GPU is in a nv-gh-runners testing pool
 gpus:
-  - 'v100'     # 40 runners
-  - 't4'       #  8 runners
-  - 'rtx2080'  #  8 runners
-  - 'rtxa6000' # 12 runners
-  - 'l4'       # 48 runners
-  - 'rtx4090'  # 10 runners
-  - 'h100'     # 16 runners
-
-# SM versions of GPUs
-gpu_sm:
-  'v100':     '70'
-  't4':       '75'
-  'rtx2080':  '75'
-  'rtxa6000': '86'
-  'l4':       '89'
-  'rtx4090':  '89'
-  'h100':     '90'
-
-# Memory size of GPUs
-gpu_mem_gb:
-  'v100':     '32'
-  't4':       '16'
-  'rtx2080':  '8'
-  'rtxa6000': '48'
-  'l4':       '24'
-  'rtx4090':  '24'
-  'h100':     '80'
-
-# GPUs that require `-testing` at the end of the runner pool name.
-testing_pool_gpus:
-  - 't4'
-  - 'rtx2080'
-  - 'rtxa6000'
-  - 'l4'
-  - 'rtx4090'
+  v100:     { sm: 70 }                # 32 GB,  40 runners
+  t4:       { sm: 75, testing: true } # 16 GB,   8 runners
+  rtx2080:  { sm: 75, testing: true } #  8 GB,   8 runners
+  rtxa6000: { sm: 86, testing: true } # 48 GB,  12 runners
+  l4:       { sm: 89, testing: true } # 24 GB,  48 runners
+  rtx4090:  { sm: 89, testing: true } # 24 GB,  10 runners
+  h100:     { sm: 90 }                # 80 GB,  16 runners
```

Comment:
minor suggestion: Instead of trying to abstract that fact that some GPUs have different runner labels, what do you think about just making `runner_label:` an explicit property of defining a GPU? I can easily foresee new runner labels getting added/changed with new structures that would break the existing logic.

Reply:
That's a good point, we may need something more flexible here at some point. We can easily deal with that when/if it becomes an issue, but I'd rather not pre-emptively change this without a usecase.